.github/workflows/terragrunt-deploy-staging-container.yml

```diff
@@ -0,0 +1,85 @@
+  TF_VAR_cloudfront_custom_header_value: ${{ secrets.STAGING_CLOUDFRONT_CUSTOM_HEADER_VALUE }}
+  TF_VAR_list_manager_endpoint: ${{ secrets.STAGING_LIST_MANAGER_ENDPOINT }}
+  TF_VAR_default_list_manager_api_key: ${{ secrets.STAGING_DEFAULT_LIST_MANAGER_API_KEY }}
+  TF_VAR_default_notify_api_key: ${{ secrets.STAGING_DEFAULT_NOTIFY_API_KEY }}
+  TF_VAR_encryption_key: ${{ secrets.STAGING_ENCRYPTION_KEY }}
+  TF_VAR_s3_uploads_bucket: ${{ secrets.STAGING_S3_UPLOADS_BUCKET }}
+  TF_VAR_s3_uploads_key: ${{ secrets.STAGING_S3_UPLOADS_KEY }}
+  TF_VAR_s3_uploads_secret: ${{ secrets.STAGING_S3_UPLOADS_SECRET }}
+  TF_VAR_c3_aws_access_key_id: ${{ secrets.STAGING_C3_AWS_ACCESS_KEY_ID }}
+  TF_VAR_c3_aws_secret_access_key: ${{ secrets.STAGING_C3_AWS_SECRET_ACCESS_KEY }}
+  TF_VAR_slack_webhook_url: ${{ secrets.STAGING_SLACK_WEBHOOK_URL }}
+  TF_VAR_wordpress_auth_key: ${{ secrets.STAGING_WORDPRESS_AUTH_KEY }}
+  TF_VAR_wordpress_secure_auth_key: ${{ secrets.STAGING_WORDPRESS_SECURE_AUTH_KEY }}
+  TF_VAR_wordpress_logged_in_key: ${{ secrets.STAGING_WORDPRESS_LOGGED_IN_KEY }}
+  TF_VAR_wordpress_nonce_key: ${{ secrets.STAGING_WORDPRESS_NONCE_KEY }}
+  TF_VAR_wordpress_auth_salt: ${{ secrets.STAGING_WORDPRESS_AUTH_SALT }}
+  TF_VAR_wordpress_secure_auth_salt: ${{ secrets.STAGING_WORDPRESS_SECURE_AUTH_SALT }}
+  TF_VAR_wordpress_logged_in_salt: ${{ secrets.STAGING_WORDPRESS_LOGGED_IN_SALT }}
+  TF_VAR_wordpress_nonce_salt: ${{ secrets.STAGING_WORDPRESS_NONCE_SALT }}
+  TF_VAR_jwt_auth_secret_key: ${{ secrets.STAGING_JWT_AUTH_SECRET_KEY }}
+  TF_VAR_wpml_site_key: ${{ secrets.STAGING_WPML_SITE_KEY }}
+
+jobs:
+  environments-manifest:
+    uses: cds-snc/gc-articles/.github/workflows/environments-manifest.yml@main
+
+  terragrunt-apply-staging:
+    needs: environments-manifest
+
+    if: |
+      (github.ref == 'refs/heads/main' && github.event_name == 'push') &&
+      (needs.environments-manifest.outputs.STAG_WORDPRESS != needs.environments-manifest.outputs.PREV_STAG_WORDPRESS)
+
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@3d8debd658c92063839bc97da5c2427100420dec # v1.3.2
+        with:
+          terraform_version: ${{ env.TERRAFORM_VERSION }}
+          terraform_wrapper: false
+
+      - name: Setup Terragrunt
+        run: |
+          mkdir bin
+          wget -O bin/terragrunt https://github.com/gruntwork-io/terragrunt/releases/download/v$TERRAGRUNT_VERSION/terragrunt_linux_amd64
+          chmod +x bin/*
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+
+      - uses: cds-snc/paths-filter@v2.10.3
+        id: filter
+        with:
+          filters: |
+            ecs:
+              - 'infrastructure/terragrunt/aws/ecs/**'
+              - 'infrastructure/terragrunt/env/staging/ecs/**'
+
+      # Load-balancer & database dependency
+      - name: Terragrunt apply ecs
```

Comment:
I think you'll end up with a double `terragrunt apply` of the ECS module here because it still exists in the current `terragrunt-apply-staging.yml` workflow:
https://github.com/cds-snc/gc-articles/blob/main/.github/workflows/terragrunt-apply-staging.yml#L135-L139

Reply:
I don't think so - I tested the triggers here (of course possible I messed up the test but I think this is right) - they don't notice when environments.yml is updated. So unless there's a change in a tf file in the ecs module, this (and the plan workflow) shouldn't be triggered. I will do a bit more testing around this to confirm.